src/uu/mv/src/mv.rs

```diff
@@ -650,139 +658,184 @@ fn rename(
-
-            let progress_bar =
-                if let (Some(multi_progress), Some(total_size)) = (multi_progress, total_size) {
-                    let bar = ProgressBar::new(total_size).with_style(
-                        ProgressStyle::with_template(
-                            "{msg}: [{elapsed_precise}] {wide_bar} {bytes:>7}/{total_bytes:7}",
-                        )
-                        .unwrap(),
-                    );
-
-                    Some(multi_progress.add(bar))
-                } else {
-                    None
-                };
-
-            #[cfg(all(unix, not(any(target_os = "macos", target_os = "redox"))))]
-            let xattrs =
-                fsxattr::retrieve_xattrs(from).unwrap_or_else(|_| std::collections::HashMap::new());
-
-            let result = if let Some(ref pb) = progress_bar {
-                move_dir_with_progress(from, to, &options, |process_info: TransitProcess| {
-                    pb.set_position(process_info.copied_bytes);
-                    pb.set_message(process_info.file_name);
-                    TransitProcessResult::ContinueOrAbort
-                })
-            } else {
-                move_dir(from, to, &options)
-            };
-
-            #[cfg(all(unix, not(any(target_os = "macos", target_os = "redox"))))]
-            fsxattr::apply_xattrs(to, xattrs)?;
-
-            if let Err(err) = result {
-                return match err.kind {
-                    fs_extra::error::ErrorKind::PermissionDenied => Err(io::Error::new(
-                        io::ErrorKind::PermissionDenied,
-                        "Permission denied",
-                    )),
-                    _ => Err(io::Error::new(io::ErrorKind::Other, format!("{err:?}"))),
-                };
-            }
+            rename_dir_fallback(from, to, multi_progress)
+        } else if is_fifo(file_type) {
+            rename_fifo_fallback(from, to)
         } else {
-            if to.is_symlink() {
-                fs::remove_file(to).map_err(|err| {
-                    let to = to.to_string_lossy();
-                    let from = from.to_string_lossy();
-                    io::Error::new(
-                        err.kind(),
-                        format!(
-                            "inter-device move failed: '{from}' to '{to}'\
-                            ; unable to remove target: {err}"
-                        ),
-                    )
-                })?;
-            }
-            #[cfg(all(unix, not(any(target_os = "macos", target_os = "redox"))))]
-            fs::copy(from, to)
-                .and_then(|_| fsxattr::copy_xattrs(&from, &to))
-                .and_then(|_| fs::remove_file(from))?;
-            #[cfg(any(target_os = "macos", target_os = "redox", not(unix)))]
-            fs::copy(from, to).and_then(|_| fs::remove_file(from))?;
+            rename_file_fallback(from, to)
         }
+    })
+}
+
+#[cfg(unix)]
+fn make_fifo(path: &Path) -> io::Result<()> {
```

Comment:
we have a similar code here:
https://github.com/uutils/coreutils/blob/1cd78c7a2b30baad4bfcf0aba5ca652b08519874/src/uu/cp/src/cp.rs#L2472

maybe we should create a function in uucore for this

Reply:
I have factored the common `make_fifo` function out into `uucore::fs`.